libcxx/include/__vector/vector.h

```diff
@@ -1031,9 +1031,14 @@ template <class _Tp, class _Allocator>
 template <class _Iterator, class _Sentinel>
```

Comment:
http://eel.is/c++draft/container.requirements#sequence.reqmts-57 says:

> T is Cpp17EmplaceConstructible into X from *i and assignable from *i[.](http://eel.is/c++draft/container.requirements#sequence.reqmts-57.sentence-1) For vector, if the iterator does not meet the forward iterator requirements ([[forward.iterators]](http://eel.is/c++draft/forward.iterators)), T is also Cpp17MoveInsertable into X[.](http://eel.is/c++draft/container.requirements#sequence.reqmts-57.sentence-2) Neither i nor j are iterators into a[.](http://eel.is/c++draft/container.requirements#sequence.reqmts-57.sentence-3)

Since we do have a requirement that T is assignable from `*i`, I think we're allowed to do this.

Can you check if we already have a benchmark that would cover this inside our vector benchmarks, and if not, add one?

Reply:
That's great, thanks a lot!